File: LeetCode/1720.py
#Link: https://leetcode.com/problems/decode-xored-array/

class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        arr = [first]
        for i in range(len(encoded)):
            arr.append(arr[i] ^ encoded[i])
        return arr
        

File: LeetCode/cpp/1720.cpp
//Link: https://leetcode.com/problems/decode-xored-array/

class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int> arr(encoded.size()+1);
        arr[0] = first;
        for(int i = 1 ; i < encoded.size() + 1 ; i++)
            arr[i] = arr[i-1] xor encoded[i-1];
        return arr;
    }
};
